### ISSUE TITLE
compatible with nestjs v8

### ISSUE THREAD

**Issue by liaoliaots** [nestjs/terminus#1316]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

npm v7 will cause conflict for now.

## Expected behavior

Let @nestjs/terminus compatible with nestjs v8.

## Environment

<pre><code>
Nest version: 8.0.0

 
For Tooling issues:
- Node version: 14.17.3  
- Platform:  Linux 

Others:

</code></pre>

**Comment by TrejGun** [comment]
<img width="484" alt="Screen Shot 2021-07-08 at 2 35 34 PM" src="https://user-images.githubusercontent.com/185285/124873998-c632c980-dff9-11eb-80d1-0db4d6da3ba4.png">

looks a bit strange in swagger

**Comment by BrunnerLivio** [comment]
@TrejGun am I missing something? Don't see something odd?

--- 


Hey all! Here is the strategy I am pursuing. Trying to finish the full migration this week

- [x] Make the library compatible with version 8 (Thank you so much @Tony133 for this)
- [ ] Release the library under the "next" distribution channel. This will allow you to already opt-in and test it upfront
- [ ] Refactor e2e tests. It's a mess + still uses the deprecated Terminus API 
- [ ] Remove deprecated APIs
- [ ] Remove unneeded peerDependencies (`@goddady/terminus`)
- [ ] Remove deprecated samples folder
- [ ] Clean-up & chore. Stricter TS / ESLint rules
- [ ] Switch from release-it to semantic-release
- [ ] Release officially!
- [ ] Update docs / migration guide

**Comment by TrejGun** [comment]
You are right. I was confused by two additional black fields on the bottom, but apparently, it is fine

**Comment by BrunnerLivio** [comment]
@TrejGun I see. These are examples you can add in Swagger. Terminus adds them out of the box - hence the two boxes :)

**Comment by BrunnerLivio** [comment]
[NestJS v8.0.0-next.0 has been released](https://www.npmjs.com/package/@nestjs/terminus/v/8.0.0-next.0) 🎉 .

```bash
npm i --save @nestjs/terminus@next
# or
npm i --save @nestjs/terminus@8.0.0-next.0
```

**THIS VERSION IS NOT FINAL. USE WITH CAUTION**. 

This pre-release still includes deprecated APIs (e.g. `TerminusModule.forRootAsync`). Be aware that these functions will get removed in the final `@nestjs/terminus@8.0.0`

**Comment by liaoliaots** [comment]
@BrunnerLivio I think that should add `@nestjs/axios` to peerDependencies or dependencies.

**Comment by BrunnerLivio** [comment]
@liaoliaots in the beginning of terminus I had these dependencies as peerDependencies. Though there were understandably some complaints around that, because if folks did not need to use the e.g. `HttpHealthIndicator` and therefore had not installed `@nestjs/axios`, npm would complain during installation. 

Still hoping for `optionalPeerDependencies` which would fix this:
https://github.com/npm/npm/issues/3066

**Comment by TrejGun** [comment]
optionalPeerDependencies is a big dream

**Comment by liaoliaots** [comment]
@BrunnerLivio I made a [minimal repo](https://github.com/liaoliaots/my-app).
I just copied the [dog](https://docs.nestjs.com/v7/recipes/terminus#custom-health-indicator) example and did not use the `HttpHealthIndicator`, which caused an error:

![2021-07-12 15-59-00屏幕截图](https://user-images.githubusercontent.com/27341089/125251633-1c6d7880-e32a-11eb-891e-7fbf33881c2d.png)

**Comment by liaoliaots** [comment]
The `HttpModule` from `@nestjs/axios` needs to be added to `imports`.

Can `HttpModule` be lazy loaded? ^-^ ^-^

**Comment by BrunnerLivio** [comment]
@liaoliaots yup planned to do so!

**Comment by khaledosman** [comment]
I'm getting this error with httpHealthCheck when using @nestjs/terminus@next

```bash
2021-07-19T11:57:07.869Z - [error] - [ExceptionHandler] - undefined is not a function - TypeError: undefined is not a function
    at HttpHealthIndicator.checkDependantPackages (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/terminus/dist/health-indicator/http/http.health.js:60:31)
    at new HttpHealthIndicator (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/terminus/dist/health-indicator/http/http.health.js:54:14)
    at Injector.instantiateClass (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:287:19)
    at callback (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:43:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Injector.resolveConstructorParams (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:115:24)
    at Injector.loadInstance (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:47:9)
    at Injector.loadProvider (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:69:9)
    at Injector.lookupComponentInImports (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:221:17)
    at Injector.lookupComponentInParentModules (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:187:33)
    at Injector.resolveComponentInstance (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:145:33)
    at resolveParam (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:99:38)
    at async Promise.all (index 1)
    at Injector.resolveConstructorParams (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:114:27)
    at Injector.loadInstance (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:47:9)
    at Injector.loadController (/Users/khaledosman/Documents/github/carbmee-api/node_modules/@nestjs/core/injector/injector.js:60:9)
```

**Comment by BrunnerLivio** [comment]
@khaledosman Ha you folks are quick on finding my experiments! 😅

Use the following version for now, that one should work.
```bash
npm i --save @nestjs/terminus@8.0.0-next.0
# For now you will also need to add @nestjs/axios
# In the official 8.0.0 it will be optional
# if you don’t use the HttpHealthIndicator
npm i —save @nestjs/axios
```

**Comment by khaledosman** [comment]
so using the same setting i'm getting a `Service Unavailable Exception`, I now had to put the full url to `pingCheck` instead of only the path

**Comment by TrejGun** [comment]
@BrunnerLivio can you please finish this?

**Comment by sgentile** [comment]
![image](https://user-images.githubusercontent.com/132404/130832084-a0c50063-1298-4620-b33c-e0d99b561b68.png)

**Comment by sgentile** [comment]
Is this an official nest module ?  Seems odd it wouldn't be compatible with nestjs common ?
